cirq/google/engine/engine.py

```diff
@@ -276,6 +277,74 @@ def run_sweep(
                                         description=job_description,
                                         labels=job_labels)
 
+    def run_batch(
+            self,
+            programs: List['cirq.Circuit'],
+            program_id: Optional[str] = None,
+            job_id: Optional[str] = None,
+            params_list: List[study.Sweepable] = None,
+            repetitions: int = 1,
+            processor_ids: Sequence[str] = ('xmonsim',),
+            gate_set: serializable_gate_set.SerializableGateSet = None,
+            program_description: Optional[str] = None,
+            program_labels: Optional[Dict[str, str]] = None,
+            job_description: Optional[str] = None,
+            job_labels: Optional[Dict[str, str]] = None,
+    ) -> engine_job.EngineJob:
+        """Runs the supplied Circuit via Quantum Engine.Creates
+
+        This will combine each Circuit provided in `programs` into
+        a BatchProgram.  Each circuit will pair with the associated
+        parameter sweep provided in the `params_list`.  The number of
+        programs is required to match the number of sweeps.
+
+        This method does not block until a result is returned.  However,
+        no results will be available until the entire batch is complete.
+
+        Args:
+            programs: The Circuits to execute as a batch.
+            program_id: A user-provided identifier for the program. This must
+                be unique within the Google Cloud project being used. If this
+                parameter is not provided, a random id of the format
+                'prog-################YYMMDD' will be generated, where # is
+                alphanumeric and YYMMDD is the current year, month, and day.
+            job_id: Job identifier to use. If this is not provided, a random id
+                of the format 'job-################YYMMDD' will be generated,
+                where # is alphanumeric and YYMMDD is the current year, month,
+                and day.
+            params_list: Parameter sweeps to use with the circuits.  The number
+                of sweeps should match the number of circuits and will be
+                paired in order with the circuits.
+            repetitions: Number of circuit repetitions to run.  Each sweep value
+                of each circuit in the batch will run with the same repetitions.
+            processor_ids: The engine processors that should be candidates
+                to run the program. Only one of these will be scheduled for
+                execution.
+            gate_set: The gate set used to serialize the circuit. The gate set
+                must be supported by the selected processor.
```

Comment:
not a fan of the default being xmon here, but we should at least document (yes run doesn't document this too)

Reply:
Changed it to raise an error, like the other parameters.